src/LayoutManager/LayoutStrategies/utils.ts

```diff
@@ -0,0 +1,56 @@
+import { Point } from '../../Point';
+import type { Group } from '../../shapes/Group';
+import type { FabricObject } from '../../shapes/Object/FabricObject';
+import type { TMat2D } from '../../typedefs';
+import { cos } from '../../util/misc/cos';
+import { multiplyTransformMatrices, qrDecompose } from '../../util/misc/matrix';
+import { calcPlaneChangeMatrix } from '../../util/misc/planeChange';
+import { degreesToRadians } from '../../util/misc/radiansDegreesConversion';
+import { sin } from '../../util/misc/sin';
+
+/**
+ * @todo this is very buggy => refactor in redo coords
+ */
+export const getObjectSizeVector = (
+  object: FabricObject,
+  /**
+   * account for selected objects in active selection
+   */
+  planeChange?: TMat2D
+) => {
+  const decomposition = planeChange
+    ? qrDecompose(
+        multiplyTransformMatrices(planeChange, object.calcOwnMatrix())
+      )
+    : undefined;
+  const sizeVector = object
+    ._getTransformedDimensions(decomposition)
+    .scalarDivide(2);
+  if (decomposition?.angle ?? object.angle) {
+    const rad = degreesToRadians(object.angle),
+      sine = Math.abs(sin(rad)),
+      cosine = Math.abs(cos(rad)),
+      rx = sizeVector.x * cosine + sizeVector.y * sine,
+      ry = sizeVector.x * sine + sizeVector.y * cosine;
```

Comment:
this is not a rotation, what kind of transformation it is?
Is a projection of what kind?

Reply:
this have to explain me, i really don't get it why this is here.

Reply:
![bbox](https://github.com/fabricjs/fabric.js/assets/34343793/4d5da73b-9452-4861-8860-1825d326e48b)

Take a look here, this is the math behind the bbox transform which takes a point from the rotated plane and transforms it to the non rotated plane.
`compenstaeAngle` which is a wrong name is an application of the bbox transform matrix

`α = object.angle`